framework/src/media/MediaPlayerImpl.cpp

```diff
@@ -74,70 +93,127 @@ player_result_t MediaPlayerImpl::destroy()
+		notifyObserver(PLAYER_OBSERVER_COMMAND_ERROR);
+		ret = PLAYER_ERROR;
+		goto exit;
+	}
+
 	mCurState = PLAYER_STATE_NONE;
-	return PLAYER_OK;
+	ret = PLAYER_OK;
+exit:
+	mSyncCv.notify_one();
 }
 
 player_result_t MediaPlayerImpl::prepare()
 {
+	player_result_t ret = PLAYER_ERROR;
+
 	std::unique_lock<std::mutex> lock(mCmdMtx);
 	medvdbg("MediaPlayer prepare\n");
-	if (mCurState != PLAYER_STATE_IDLE) {
-		meddbg("MediaPlayer prepare fail : wrong state\n");
+
+	PlayerWorker& mpw = PlayerWorker::getWorker();
+	if (!mpw.isAlive()) {
 		return PLAYER_ERROR;
 	}
 
-	PlayerWorker& mpw = PlayerWorker::getWorker();
-	mpw.getQueue().enQueue(&MediaPlayerImpl::notifySync, shared_from_this());
+	mpw.getQueue().enQueue(&MediaPlayerImpl::preparePlayer, shared_from_this(), std::ref(ret));
 	mSyncCv.wait(lock);
 
+	return ret;	
+}
+
+void MediaPlayerImpl::preparePlayer(player_result_t &ret)
+{
+	medvdbg("MediaPlayer Worker : preparePlayer\n");
+	if (mCurState != PLAYER_STATE_IDLE) {
+		meddbg("MediaPlayer prepare fail : wrong state\n");
+		notifyObserver(PLAYER_OBSERVER_COMMAND_ERROR);
+		ret = PLAYER_ERROR;
+		goto exit;
+	}
 	if (mInputDataSource == nullptr) {
 		meddbg("MediaPlayer prepare fail : mInputDataSource is not set\n");
-		return PLAYER_ERROR;
+		notifyObserver(PLAYER_OBSERVER_COMMAND_ERROR);
+		ret = PLAYER_ERROR;
+		goto exit;
 	}
 
 	if (!mInputDataSource->open()) {
 		meddbg("MediaPlayer prepare fail : file open fail\n");
-		return PLAYER_ERROR;
+		notifyObserver(PLAYER_OBSERVER_COMMAND_ERROR);
+		ret = PLAYER_ERROR;
+		goto exit;
 	}
 
 	if (set_audio_stream_out(mInputDataSource->getChannels(), mInputDataSource->getSampleRate(),
 							 mInputDataSource->getPcmFormat()) != AUDIO_MANAGER_SUCCESS) {
 		meddbg("MediaPlayer prepare fail : set_audio_stream_out fail\n");
-		return PLAYER_ERROR;
+		notifyObserver(PLAYER_OBSERVER_COMMAND_ERROR);
+		ret = PLAYER_ERROR;
+		goto exit;
 	}
 
 	mBufSize = get_output_frames_byte_size(get_output_frame_count());
 	if (mBufSize < 0) {
 		meddbg("MediaPlayer prepare fail : get_output_frames_byte_size fail\n");
-		return PLAYER_ERROR;
+		notifyObserver(PLAYER_OBSERVER_COMMAND_ERROR);
+		ret = PLAYER_ERROR;
+		goto exit;
 	}
 
 	medvdbg("MediaPlayer mBuffer size : %d\n", mBufSize);
 
 	mBuffer = new unsigned char[mBufSize];
 	if (!mBuffer) {
 		meddbg("MediaPlayer prepare fail : mBuffer allocation fail\n");
-		return PLAYER_ERROR;
+		notifyObserver(PLAYER_OBSERVER_COMMAND_ERROR);
+		ret = PLAYER_ERROR;
+		goto exit;
 	}
 
 	mCurState = PLAYER_STATE_READY;
-	return PLAYER_OK;
+	ret = PLAYER_OK;
+exit:
+	mSyncCv.notify_one();
 }
 
 player_result_t MediaPlayerImpl::unprepare()
 {
+	player_result_t ret = PLAYER_ERROR;
+
 	std::unique_lock<std::mutex> lock(mCmdMtx);
 	medvdbg("MediaPlayer unprepare\n");
-	if (mCurState == PLAYER_STATE_NONE || mCurState == PLAYER_STATE_IDLE) {
-		meddbg("MediaPlayer unprepare fail : wrong state\n");
+
+	PlayerWorker& mpw = PlayerWorker::getWorker();
+	if (!mpw.isAlive()) {
```

Comment:
return PLAYER_ERROR will raise svace issue, you can return just ret.

Reply:
Fixed it